torch/_inductor/runtime/static_cuda_launcher.py

```diff
@@ -44,10 +44,15 @@ def __init__(self, kernel: CompiledKernel) -> None:
         self.declared_constexprs = kernel.src.fn.constexprs
 
         self.hash = kernel.hash
-        if (
-            kernel.__class__.launch_enter_hook is not None
-            or kernel.__class__.launch_exit_hook is not None
-        ):
+
+        if knobs is None:
```

Comment:
nit: knobs as a name for "triton configuration" is a bit short; might be better to just put torch._indcutor.runtime.triton_compat.knobs as a full name here instead